### ISSUE TITLE
-D selection missing absolute value step

### ISSUE THREAD

**Issue by jidanni** [GenericMappingTools/gmt#7716]
Here the automatic -D selection should be the same.

I suspect a missing absolute value step.

```
gmt begin
gmt coast -V -R-90/20/-65/30+r -JOB-80/25/-70/20/1i -W -B
# coast [INFORMATION]: Map scale is 680.545 km per cm or 1:6.80545e+07.
# coast [INFORMATION]: -D option: Selected resolution -Dl
gmt coast -V -R-90/20/-65/30+r -JOB-70/20/-80/25/1i -W -B
# coast [INFORMATION]: Map scale is -680.545 km per cm or 1:-6.80545e+07.
# coast [INFORMATION]: -D option: Selected resolution -Df
gmt end show
```
So -680.545 needs to be converted to 680.545 before the -D is calculated.

(Actually, with -JOB, going from point 1 to point 0 should just
flip the map upside down, and not make a "negative scale." But I'm
old fashioned.) Version 6.4.

**Comment by joa-quim** [comment]
I get only positive scales with GMTdev, but result is still -Df, which seems correct.
```
gmt pscoast -V -R-90/20/-65/30+r -JOB-70/20/-80/25/1i -W -B -Da -P > lixo.ps
pscoast [INFORMATION]: GMT_Parse_Options: Interval-setting -B options were reordered to appear before axis and frame -B options to ensure proper parsing.
pscoast [INFORMATION]: GMT_Parse_Options: New option order: -V -R-90/20/-65/30+r -JOB-70/20/-80/25/1i -W -Da -P -Baf -BWSen
pscoast [INFORMATION]: Spherical approximation used
pscoast [INFORMATION]: Auto-frame interval for x-axis (item 0): a5f1
pscoast [INFORMATION]: Auto-frame interval for y-axis (item 0): a2f1
pscoast [INFORMATION]: gmt_map_setup perimeter search region: 269.9/295.1000000000003/13.27574581879438/37.67901117118037.
pscoast [INFORMATION]: Map scale is 680.545 km per cm or 1:6.80545e+07.
pscoast [INFORMATION]: -D option: Selected resolution -Df
```
What I find more puzzling (though possible) is that both -JOB gaive the same projected points
```
echo -80 30 | gmt mapproject -R-90/20/-65/30+r -JOB-70/20/-80/25/1i
0.678573438445  2.10186530455

echo -80 30 | gmt mapproject -R-90/20/-65/30+r -JOB-80/25/-70/20/1i
0.678573438445  2.10186530455
```

**Comment by joa-quim** [comment]
Too small of a map possibly. Computing the projected coords show they are different.
```
echo -80 30 | gmt mapproject -R-90/20/-65/30+r -Job-80/25/-70/20/1i -C -F
-248007.641772  498227.750891

echo -80 30 | gmt mapproject -R-90/20/-65/30+r -Job-70/20/-80/25/1i -C -F
-1415510.16331  498227.750891
```

**Comment by joa-quim** [comment]
Actually you were right. Despite dev printing a positive scale internally, at the time of resolution selection, it was still negative. Fixed by #7739.
